README.md

```diff
@@ -6,27 +6,392 @@
+
+[Poetry]: https://python-poetry.org/
+[dependabot]: https://github.com/dependabot
+
+However, if you want to update them manually, you can do so by running:
+
+```shell
+poetry update
+```
+
+This will update all dependencies according to the rules defined in `pyproject.toml`.
+To see all outdated dependencies, run:
+
+```shell
+poetry show -o
+```
+
+Note that this will also include any sub-dependencies that are not directly defined in `pyproject.toml`.
+
+### Background tasks
+
+Scheduled & background tasks are run with [Celery].
+
+[Celery]: https://github.com/celery/
+
+When developing locally, you can run these tasks in a Celery worker with `make celery`.
+This uses the filesystem as the message broker.
+You'll need to create queue and processed folders according to the
+`CELERY_QUEUE_FOLDER_OUT`, `CELERY_QUEUE_FOLDER_IN`, `CELERY_PROCESSED_FOLDER`
+environment variables (see `.env.example`).
+
+If you want to run background tasks synchronously without Celery, set the environment variable
+`CELERY_ENABLED` to `False`. Scheduled tasks still need the worker in order to run.
+
+### Authentication
+
+Authentication is handled by [Helsinki Tunnistus] Keycloak using the [django-helusers] library.
+You'll need to get the `TUNNISTAMO_ADMIN_SECRET` from the [Azure Pipelines library]
+or from a colleague and set that in your `.env` file.
+
+[Helsinki Tunnistus]: https://helsinkisolutionoffice.atlassian.net/wiki/spaces/KAN/pages/419463195/
+[django-helusers]: https://github.com/City-of-Helsinki/django-helusers
+[Azure Pipelines library]: https://dev.azure.com/City-of-Helsinki/tilavarauspalvelu/_library?itemType=VariableGroups
+
+Authentication in application is managed using session cookies.
+
+### Static files
+
+Static files are served by the [Whitenoise] package.
+These are all files that are not uploaded by the users in Django Admin pages.
+
+[Whitenoise]: https://whitenoise.evans.io/en/stable/
+
+Media files are served by the [uWSGI static files implementation], offloaded to threads.
+These are all files uploaded by users in Django Admin pages.
+
+[uWSGI static files implementation]: https://uwsgi-docs.readthedocs.io/en/latest/StaticFiles.html
+
+### Image cache
+
+In production, [Varnish cache] is used for reservation unit and purpose images.
+When new image is uploaded, existing images are removed from the cache.
+
+[Varnish cache]: https://varnish-cache.org/
+
+In settings there are four configurations:
+- `IMAGE_CACHE_ENABLED` = Toggle caching on/off
+- `IMAGE_CACHE_VARNISH_HOST` = Varnish hostname
+- `IMAGE_CACHE_PURGE_KEY` = Secret key for doing purge requests
+- `IMAGE_CACHE_HOST_HEADER` = `Host` header value in purge request
+
+### Translations
+
+Translations are handled by Django's built-in translation system.
+GitHub Actions CI will check that all translations are up-to-date during PRs.
+To update translations, run `make translations`. This will update the `.po` files
+located in the `locale` directory.
```

Comment:
If new translations are added or old translatable strings changed, run `make translations` to automatically update the `.po` files
After updating the translations to `.po` files, run `make translate` to compile `.mo` files which are actually used by Django for localisation (only needed to use the translations locally, pipeline does this for the servers)